cst438_gradebook/src/main/java/com/cst438/services/RegistrationServiceMQ.java

```diff
@@ -44,16 +44,23 @@ public RegistrationServiceMQ() {
 	@Transactional
 	public void receive(EnrollmentDTO enrollmentDTO) {
 		
-		//TODO  complete this method in homework 4
-		
+	   System.out.println("Recieved Rabbit message: "+enrollmentDTO);
+	   Course crs = courseRepository.findById(enrollmentDTO.course_id).orElse(null);
+      Enrollment enrollment = new Enrollment();
+      enrollment.setStudentEmail(enrollmentDTO.studentEmail);
+      enrollment.setStudentName(enrollmentDTO.studentName);
+      enrollment.setCourse(crs);
+      enrollmentRepository.save(enrollment);
 	}
 
 	// sender of messages to Registration Service
 	@Override
 	public void sendFinalGrades(int course_id, CourseDTOG courseDTO) {
 		 
 		//TODO  complete this method in homework 4
-		
+	   System.out.println("Sending grades for course " + course_id + " to register MQ service");
+	   rabbitTemplate.convertAndSend(registrationQueue.getName(), courseDTO);
```

Comment:
Should we set course_id for the courseDTO object here? I don't think it knows which course to update grades for...

Reply:
I believe the course_id is set in the API
![githubweek4](https://user-images.githubusercontent.com/74542642/170905766-cabcbf97-5bb6-4590-a825-e784262ae124.PNG)

Reply:
I think RegistrationServiceREST.java makes use of the course id